libs/journeys/ui/src/components/SearchBar/SearchDropdown/CountryLanguageSelector/CountryLanguageSelector.tsx

```diff
@@ -0,0 +1,128 @@
+import Box from '@mui/material/Box'
+import Chip from '@mui/material/Chip'
+import Stack from '@mui/material/Stack'
+import Typography from '@mui/material/Typography'
+import { RefinementListRenderState } from 'instantsearch.js/es/connectors/refinement-list/connectRefinementList'
+import Image from 'next/image'
+import { ReactElement, useEffect, useState } from 'react'
+
+import { getTopSpokenLanguages } from '../../../../libs/algolia/getTopSpokenLanguages'
+import { useSearchBar } from '../../../../libs/algolia/SearchBarProvider'
+import { useCountryQuery } from '../../../../libs/useCountryQuery'
+
+// The cookie is set by the watch middleware
+// Only use for tests
+export const NEXT_COUNTRY = 'NEXT_COUNTRY=00001---US'
+
+interface CountryLanguageSelectorProps {
+  refinements: RefinementListRenderState
+}
+
+export function CountryLanguageSelector({
+  refinements
+}: CountryLanguageSelectorProps): ReactElement {
+  const { items, refine } = refinements
+
+  const { dispatch } = useSearchBar()
+  const [country, setCountry] = useState<string>()
+  const [countryCode, setCountryCode] = useState<string>()
+  const { data } = useCountryQuery({ countryId: countryCode ?? '' })
+
+  const spokenLanguages = getTopSpokenLanguages({
+    country: data?.country,
+    availableLanguages: items
+  })
+
+  function handleClick(language: string): void {
+    const continent = data?.country?.continent?.name?.find(Boolean)?.value
+    const refinedItem = items.find((item) => item.label === language)?.isRefined
+
+    if (continent == null) return
+
+    refine(language)
+
+    if (refinedItem === true) {
+      dispatch({
+        type: 'RemoveLanguageContinents',
+        language
+      })
+    } else {
+      dispatch({
+        type: 'SelectLanguageContinent',
+        continent,
+        language,
+        isSelected: true
+      })
+    }
+  }
+
+  useEffect(() => {
+    const countryCookie = document.cookie
+      .split('; ')
+      .find((row) => row.startsWith('NEXT_COUNTRY'))
+      ?.split('=')[1]
+
+    if (countryCookie != null) {
+      const [, countryCode] = countryCookie.split('---')
+      const displayNames = new Intl.DisplayNames(['en'], { type: 'region' })
+      const countryName = displayNames.of(countryCode)
+      setCountry(countryName)
+      setCountryCode(countryCode)
+    }
+  }, [])
```

Comment:
Would be cleaner to refactor this out.

```suggestion
  ...
  
  function findUserCountry() {
    const countryCookie = getCountryCookie()
      if (countryCookie != null) {
        const countryName = parseCountryName(countryCookie)
        setCountry(countryName)
        setCountryCode(countryCode)
      }
    }
  }
  
  useEffect(findUserCountry, [])
```